docker-compose.yml

```diff
@@ -1,64 +1,64 @@
-version: "2.1"
-services:
-    zookeeper:
-        image: wurstmeister/zookeeper
-        ports:
-            - "2181:2181"
-    kafka-broker:
-        image: wurstmeister/kafka:2.12-2.0.1
-        ports:
-            - "9092:9092"
-        environment:
-            HOSTNAME_COMMAND: "route -n | awk '/UG[ \t]/{print $$2}'"
-            KAFKA_CREATE_TOPICS: "names:1:1,test:1:1,greetings:1:1"
-            KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
-        depends_on:
-            - zookeeper
-        volumes:
-            - /var/run/docker.sock:/var/run/docker.sock
-    master:
-        build:
-            context: .
-        expose:
-            - "6123"
-        ports:
-            - "8081:8081"
-        environment:
-            - ROLE=master
-            - MASTER_HOST=master
-        volumes:
-            - ./checkpoint-dir:/checkpoint-dir
-    worker:
-        build:
-            context: .
-        expose:
-            - "6121"
-            - "6122"
-        depends_on:
-            - master
-            - kafka-broker
-        links:
-            - "master:master"
-            - "kafka-broker:kafka-broker"
-        environment:
-            - ROLE=worker
-            - MASTER_HOST=master
-        volumes:
-            - ./checkpoint-dir:/checkpoint-dir
-    function-endpoint:
-        build:
-            context: ./statefun
-        expose:
-            - "8000"
-        ports:
-            - "8000:8000"
-    api-endpoint:
-        build:
-            context: ./endpoints
-        depends_on: 
-            - kafka-broker
-        expose:
-            - "5000"
-        ports:
-            - "5000:5000"
-        
+version: "2.1"
+services:
+    zookeeper:
+        image: wurstmeister/zookeeper
+        ports:
+            - "2181:2181"
+    kafka-broker:
+        image: wurstmeister/kafka:2.12-2.0.1
+        ports:
+            - "9092:9092"
+        environment:
+            HOSTNAME_COMMAND: "route -n | awk '/UG[ \t]/{print $$2}'"
+            KAFKA_CREATE_TOPICS: "users:1:1,users-create:1:1,user-events:1:1,names:1:1"

```

Comment:
Right now we create 1 partition per topic. For now thats fine, but later on we might want to increase to improve performance. Also if you use a consumer group reading from a topic with one partition only one consumer will be able to read.  So then if we only have one partition and a set of (scaled) stateful functions, only one stateful function will actually do stuff. Anyhow, we should not forget to change this later.

Reply:
Yes yes, I left that for later but we can do some performance improvements there!